### ISSUE TITLE
Runtime error during ibdsim. Slice bounds out of range.

### ISSUE THREAD

**Issue by kcalvinalvin** [mit-dci/utreexo#12]
Log:
```
./ibdsim -ttlfn ttl.testnet.txos
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
main.runIBD(0x0, 0x0)
        /home/calvin/go/src/github.com/mit-dci/utreexo/cmd/ibdsim/ibdsim.go:135 +0x113c
main.main()
        /home/calvin/go/src/github.com/mit-dci/utreexo/cmd/ibdsim/ibdsim.go:43 +0xbb
```

On line 76 inside `func runIBD()`, `var scheduleBuffer` is declared and the length is set to 0 but at line 135, there is a slice to this, resulting in `slice bounds out of range` error.

https://github.com/mit-dci/utreexo/blob/71f49e579f8770c79f4fe28b62b8ab58d07a7c67/cmd/ibdsim/ibdsim.go#L76

https://github.com/mit-dci/utreexo/blob/71f49e579f8770c79f4fe28b62b8ab58d07a7c67/cmd/ibdsim/ibdsim.go#L135

There are chunks of code commented out that uses `var scheduleBuffer`. I'm guessing you were testing something but haven't got around to it yet. 

Is `clair` binary also needed during the ibdsim at the moment? There is a comment about clair inside the aforementioned commented out chunk 

https://github.com/mit-dci/utreexo/blob/71f49e579f8770c79f4fe28b62b8ab58d07a7c67/cmd/ibdsim/ibdsim.go#L99-L108

Also, maybe you could delete

https://github.com/mit-dci/utreexo/blob/71f49e579f8770c79f4fe28b62b8ab58d07a7c67/cmd/ibdsim/ibdsim.go#L130

since 

https://github.com/mit-dci/utreexo/blob/71f49e579f8770c79f4fe28b62b8ab58d07a7c67/cmd/ibdsim/ibdsim.go#L132-L136

**Comment by adiabat** [comment]
Oops!  Yeah that was leftover from "clair", which I gave up on as it didn't seem to give the performance improvement I hoped for (but maybe I was doing it wrong?)
Should be fixed in 721bd7dd1adacbfc0d21c9eb4c240e5760486da1 
(Probably should clean out the commented out lines, but maybe clair could be re-explored...)

**Comment by kcalvinalvin** [comment]
Maybe keep it in another branch?
